docs/source/features/mixed_precision.rst

```diff
@@ -13,8 +13,49 @@ NeMo Framework supports half-precision FP16 and BF16 computation training via Me
 This training recipe uses half-precision in all layer computation keeping the model states (optimizer states and master parameters) in single-precision.
 To avoid repeated data type casting at each layer computation, Megatron Core keeps a separate copy of half-precision parameters that is updated after each optimizer step.
 
-Half-precision training is enabled when setting ``precision`` to either of ``fp16-mixed`` or ``bf16-mixed`` along with  ``megatron_amp_O2=true``.
-The parameter gradients are computed in the same half-precision, and the precision of gradient reduce-scatter across data-parallel GPUs can be set by ``optim.grad_sync_dtype``.
+Half-precision training is enabled when setting trainer's ``plugins`` to either of ``fp16-mixed`` or ``bf16-mixed``.
+The parameter gradients are computed in the same half-precision, and the precision of gradient reduce-scatter across data-parallel GPUs is set automatically according to the trainer's precision.
+
+Usage
+^^^^^
+
+.. code-block:: python
+
+  import nemo_run as run
+
+  from nemo import lightning as nl
+  from nemo.collections.llm.recipes.precision.mixed_precision import bf16_mixed, fp16_mixed
+  
+  trainer_args = {TRAINER_ARGS}
+
+  # Set up trainer with bf16 precision
+  trainer_bf16 = run.Config(
+    nl.Trainer,
+    plugins=bf16_mixed(),
+    **trainer_args,
+  )
+
+  # Set up trainer with fp16 precision
+  trainer_fp16 = run.Config(
+    nl.Trainer,
+    plugins=fp16_mixed(),
+    **trainer_args,
+  )
+
+It's also possible to change precision for a specific recipe:
+
+.. code-block:: python
+
+  from functools import partial
+
+  from nemo.collections import llm
+  from nemo.collections.llm.recipes.precision.mixed_precision import bf16_mixed, fp16_mixed
+
+  # Load recipe
+  recipe = partial(llm.llama3_8b.pretrain_recipe)()
+
+  # Change precision
+  recipe.trainer.plugins = fp16_mixed()
 
 FP8 Training
 ============
```

Comment:
Line 22/63 fix punctuation "F8-related arguments"

NVIDIA H100 GPU introduced support for a new datatype, FP8 (8-bit floating point), enabling higher throughput of matrix multiplies and convolutions. NeMo Framework uses the NVIDIA [TransformerEngine](https://github.com/NVIDIA/TransformerEngine) (TE) to leverage speedups from FP8. The following table summarizes the FP8-related arguments that can be configured in NeMo ([example config setting](https://github.com/NVIDIA/NeMo/blob/2e1814c9f031ad2aeeebad44597365e97253d2c4/examples/nlp/language_modeling/conf/megatron_gpt_config.yaml/#L192-L200)). For a more detailed overview, refer to the TE [documentation](https://docs.nvidia.com/deeplearning/transformer-engine/user-guide/index.html), specifically the FP8 [format](https://docs.nvidia.com/deeplearning/transformer-engine/user-guide/api/common.html#transformer_engine.common.recipe.Format) and [recipe](https://docs.nvidia.com/deeplearning/transformer-engine/user-guide/api/common.html#transformer_engine.common.recipe.DelayedScaling).